source/fundamentals/class-mapping.txt

```diff
@@ -0,0 +1,328 @@
+a class map.
+
+Ignore Extra Elements
+~~~~~~~~~~~~~~~~~~~~~
+
+When a BSON document is deserialized to a {+language+} class, the {+driver-short+}
+looks at the name of each field in the document and tries to find a matching property
+name in the class. By default, if a field in the document doesn't have a match in the class,
+the driver throws an exception. 
+
+You can choose to ignore any elements that do 
+not have a matching class property by using the ``BsonIgnoreExtraElements`` attribute.
+This prevents the driver from throwing an exception, and maps any other fields
+that have matching class properties.
+
+The following example shows how to add a ``BsonIgnoreExtraElements`` attribute
+to a class.
+
+.. code-block:: csharp
+
+   [BsonIgnoreExtraElements]
+   public class Person
+   {
+       public string Name { get; set; }
+       public int Age { get; set; }
+       public List<string> Hobbies {get; set;}
+   }
+
+You can also ignore any extra elements when registering a class map:
+
+.. code-block:: csharp
+   
+   BsonClassMap.RegisterClassMap<Person>(classmap => 
+   {
+        classmap.AutoMap();
+        classmap.SetIgnoreExtraElements(true);
+   });
+
+Using Class Discriminators
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+You can specify **discriminators** to help identify **polymorphic** classes that
+are serialized to the same collection. Polymorphic classes are classes that
+inherit properties and methods from a parent class. A discriminator is an
+element that's added to a document to identify which class the document was
+serialized from.
+
+You can specify a discriminator using the ``BsonDiscriminator`` attribute as
+follows:
+
+.. code-block:: csharp
+
+   [BsonDiscriminator("personClass")]
+   public class Person 
+   {
+       public string Name { get; set; }
+       public int Age { get; set; }
+       public List<string> Hobbies {get; set;}
+   }
+
+You can also specify a discriminator when registering a class map as follows:
+
+.. code-block:: csharp
+
+   BsonClassMap.RegisterClassMap<Person>(classmap => 
+   {
+       classmap.AutoMap();
+       classmap.SetDiscriminator("personClass");
+   });
+
+In BSON, discriminators have the field name ``_t``. 
+
+The following example shows how a document from the ``Person`` class with the
+"personClass" discriminator appears in the collection after serialization:
+
+.. code-block:: json
+
+   { "_id": "...", "_t": "personClass", "name": "...", "age": "...", "hobbies": [...]}
+
+.. TODO: Link to page on polymorphism/discriminators
+
+Mapping with Constructors
+-------------------------
+
+By default, the {+driver-short+} can automatically map a class only if the class has
+a constructor with no arguments. If you want the driver to use a constructor that accepts
+one or more arguments, you can add the ``BsonConstructor`` attribute to the constructor.
+In this case, the driver parses the expression tree to determine how to map the   
+constructor arguments to class properties or fields.
```

Comment:
"parses the expression tree" is not an accurate statement

"examines the type using reflection" is more accurate

Reply:
Good to know. Would it be sufficient to say "the driver examines the types to determine how to map the   
constructor arguments to class properties or fields", to avoid confusing anyone not familiar with reflection? 

I could also mention reflection and link to more information about it if we think it's necessary to mention.

Reply:
Examines the types is a good way of putting it.

Reply:
Are you going to make this change?

Reply:
Oops, thought I did that earlier. It's changed now!